azext_iot/_params.py

```diff
@@ -164,25 +165,58 @@ def load_arguments(self, _):
     with self.argument_context('iot dps enrollment create') as context:
         context.argument('attestation_type', options_list=['--attestation-type', '-at'],
                          arg_type=get_enum_type(AttestationType), help='Attestation Mechanism')
-        context.argument('certificate_path', options_list=['--certificate-path', '-cp'],
-                         help='The path to the file containing the certificate. '
-                         'When choosing x509 as attestation type, certificate path is required.')
+        context.argument('certificate_path',
+                         options_list=['--certificate-path', '-cp'],
+                         help='The path to the file containing the primary certificate. '
+                         'When choosing x509 as attestation type, '
+                         'one of the certificate path is required.')
+        context.argument('secondary_certificate_path',
+                         options_list=['--secondary-certificate-path', '-scp'],
+                         help='The path to the file containing the secondary certificate. '
+                         'When choosing x509 as attestation type, '
+                         'one of the certificate path is required.')
         context.argument('endorsement_key', options_list=['--endorsement-key', '-ek'],
                          help='TPM endorsement key for a TPM device. '
-                         ' When choosing tpm as attestation type, endorsement key is required.')
+                         'When choosing tpm as attestation type, endorsement key is required.')
 
     with self.argument_context('iot dps enrollment update') as context:
-        context.argument('certificate_path', options_list=['--certificate-path', '-cp'],
-                         help='The path to the file containing the certificate. '
-                         'When updating enrollment with x509 attestation mechanism, '
-                         'certificate path is required.')
+        context.argument('certificate_path',
```

Comment:
It looks like you can put '--certificate-path', '--secondary-certificate-path', '--remove-certificate' and '--remove-secondary-certificate' in the parent 'iot dps' argument context to reduce repetition.

Reply:
I will put '--remove-certificate' and '--remove-secondary-certificate' in the parent, but '--certificate-path', '--secondary-certificate-path' would have to stay because it differs between enrollment, enrollment-group as well as between create and update.

Reply:
Ok. Is that because of the difference in help text?

Reply:
Yea it should work like that

Reply:
Yes. Whether Enrollment cert path is required is based on its attestation type. Group enrollment cert path is always required. When update, none of them is required, and that is what I put into the parent since they are common.